MarkdownFormattingToHTML

chfsx opened the pull request:
@klees this adds a new Transformation `MarkdownFormattingToHTML` which uses the library we decided to use to convert Markdown to HTML. This case is certainly an excellent example that can be converted within the framework of https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/docs/development/text-handling.md. For the moment, however, it is important to offer this function centrally in order to be able to find all of its places.

chfsx committed:
MarkdownFormattingToHTML

mjansenDatabay commented on tests/Refinery/String/MarkdownFormattingToHTMLTest.php:
```diff
@@ -0,0 +1,70 @@
+
+    public function stringProvider(): array
+    {
+        // Never ever try to change something on this array :-) e.g. a 'ä' isn't a 'ä' but a 'ä' ;-)
```
Could you please provide more details here? I don't get it :-).

mjansenDatabay commented on tests/Refinery/String/MarkdownFormattingToHTMLTest.php:
```diff
@@ -0,0 +1,70 @@
+{
+    private Transformation $markdown;
+
+    public function setUp(): void
```
```suggestion
    protected function setUp(): void
```

mjansenDatabay reviewed (changes requested):
Hi @chfsx ,

thank you a lot for contributing to the `Refinery`.

Please answer the following questions:
- [x] What do you mean by `a 'ä' isn't a 'ä' but a 'ä' ;-)`?

Please implement the following changes:
- [x] Fix the access modifier of the overwritten `setUp` modifier in the unit test.

You already addressed the other topic I would have raised, namely the correlation to Richard's document.

Best regards,
@mjansenDatabay

chfsx commented:
Hi @mjansenDatabay, thanks for the feedback!
> a 'ä' isn't a 'ä' but a 'ä' ;-)
this was a copy/paste mistake from https://github.com/ILIAS-eLearning/ILIAS/blob/trunk/tests/Refinery/String/UTFNormalTest.php :-) I removed the line
and changed the access modifier of `setUp`
